internal/core/v1beta1/distribution_service.go

```diff
@@ -66,20 +67,32 @@ func (d *DistributionService) CreateInvalidation(ctx context.Context, distributi
 		return nil, err
 	}
 
-	// prepend prefix to all the paths
-	absolutePaths := make([]string, 0, len(paths))
+	cleanedPaths := make([]string, 0, len(paths))
+	invalidPaths := make([]string, 0)
+
 	for _, p := range paths {
-		// prevent users from doing funny business by going up a directory
-		if strings.Contains(p, "../") {
-			return nil, NewInvalidationError(BadRequestErrorCode, fmt.Errorf("invalid path"), errors.New("path cannot contain ../"))
+		// URL decode the path, and clean it
+		decodedPath, err := url.PathUnescape(p)
+		if err != nil {
+			return nil, NewInvalidationError(BadRequestErrorCode, errors.New("invalid encoded path"), fmt.Errorf("invalid encoded path: %v", p))
```

Comment:
I don't think we should decode the path here.
- From this [comment](https://github.com/aws/aws-sdk-go/issues/223#issuecomment-100535238), Cloudfront takes the paths as literal strings and does not interpret them. So attacks like `/foo/%2e%2e%2f/*` should have no effect.
- By expecting valid encoded paths, we're restricting users from having actual symbols like `ab%2` in their paths as demonstrated by the test in distribution_service_test.go [L199](https://github.com/kanopy-platform/cdnvalidator/pull/20/files#diff-7212ce90765de18c46c2510155b75b6287ec4c030d8cf2d15962958751586f8dR199)

I think it should be okay to do only a filepath.Clean and ensure the prefix exists.

Reply:
That sounds good.  We just need to make sure that a path attack won't get around our entitlement layer.